bigchaindb/models.py

```diff
@@ -59,7 +59,7 @@ def validate(self, bigchain, current_transactions=[]):
 
                 spent = bigchain.get_spent(input_txid, input_.fulfills.output,
                                            current_transactions)
-                if spent and spent.id != self.id:
+                if spent:
```

Comment:
This is a more strict condition. IMHO there is no need to match ids i.e. if there a txn which spends the current input then regardless of the transaction id it should be considered a double spend.

Reply:
Yes, because we also check to see if the current transaction (self) has more than one input trying to spend the same output (below). Actually, the line:

`raise DoubleSpend('tx "{}" spends inputs twice'.format(self.id))`

should say

`raise DoubleSpend('tx "{}" spends the same output more than once'.format(self.id))`

Maybe make that change in this PR as well, because it helps to make the logic more clear.